File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;

class Solution {

    public int solution(int[][] routes) {
        int answer = 0;
        int lastPos = -30001;
        Arrays.sort(routes, Comparator.comparingInt(a->a[1]));
        for(int [] arr : routes) {
            if(lastPos < arr[0]){
                lastPos = arr[1];
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <set>
#include <unordered_set>
using namespace std;

int solution(vector<vector<int>> routes) {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int answer = 0;
    int lastPos = -30001;
    sort(routes.begin(), routes.end(), [&](const vector<int> &a, const vector<int> &b)
    {
        return a[1] < b[1];
    });

    for (auto route : routes)
    {
        if (lastPos < route[0])
        {
            lastPos = route[1];
            answer++;
        }
    }
    return answer;
}